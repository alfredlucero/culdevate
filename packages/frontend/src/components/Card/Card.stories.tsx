import React from "react";
import { storiesOf } from "@storybook/react";
import Card from "./index";

storiesOf("Common/Card", module).add("Default", () => (
  <div style={{ width: "400px", padding: "30px", backgroundColor: "#efefef" }}>
    <Card>Card Children</Card>
  </div>
));
