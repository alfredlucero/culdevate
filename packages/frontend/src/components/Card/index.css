.card {
  @apply bg-white shadow-md rounded px-8 py-6;
}
