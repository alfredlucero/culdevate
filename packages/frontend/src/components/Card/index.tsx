import React from "react";
import cn from "classnames";
import { CommonProps } from "../commonProps";

interface CardProps extends CommonProps {
  children: React.ReactNode;
}

const Card: React.FC<CardProps> = ({ children, className = "", testId = "", ...passThroughProps }) => {
  return (
    <div
      className={cn("bg-white", "shadow-md", "rounded", "px-8", "py-6", className)}
      {...(testId !== "" ? { "data-testid": testId } : {})}
      {...passThroughProps}
    >
      {children}
    </div>
  );
};

export default Card;
