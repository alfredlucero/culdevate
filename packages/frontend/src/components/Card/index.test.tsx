import React from "react";
import { render } from "@testing-library/react";
import Card from "./index";

describe("<Card />", () => {
  test("should render without error", () => {
    const { container } = render(
      <Card className="extra-card-class" testId="someCardTestId">
        <p>Children</p>
      </Card>,
    );

    expect(container).toMatchSnapshot();
  });
});
