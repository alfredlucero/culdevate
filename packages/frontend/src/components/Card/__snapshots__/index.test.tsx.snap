// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render without error 1`] = `
<div>
  <div
    class="bg-white shadow-md rounded px-8 py-6 extra-card-class"
    data-testid="someCardTestId"
  >
    <p>
      Children
    </p>
  </div>
</div>
`;
